proof_ipa.go

```diff
@@ -524,20 +595,28 @@ func PostStateTreeFromStateDiff(preroot VerkleNode, statediff StateDiff) (Verkle
 
 	for _, stemstatediff := range statediff {
 		var (
-			values     = make([][]byte, NodeWidth)
-			overwrites bool
+			values    = make([][]byte, NodeWidth)
+			overwrite bool
 		)
 
-		for _, suffixdiff := range stemstatediff.SuffixDiffs {
-			if /* len(suffixdiff.NewValue) > 0 - this only works for a slice */ suffixdiff.NewValue != nil {
+		for i, suffix := range stemstatediff.InsertedSuffixes {
+			if /* len(suffixdiff.NewValue) > 0 - this only works for a slice */ stemstatediff.InsertedNew[i] != nil {
```

Comment:
I'm a bit lost here. Shouldn't `stemstatediff.InsertedNew` items always be not nil?
If was an inserted value, we need a value.